pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -378,28 +393,49 @@ func getPacketFile(filePath string) (afero.File, error) {
 	return file, nil
 }
 
-// getTargetCaptureDevice is trying to locate the target device for packet capture. If the target
+// getTargetCaptureDevices is trying to locate the target devices for packet capture. If the target
 // Pod does not exist on the current Node, the agent on this Node will not perform the capture.
 // In the PacketCapture spec, at least one of `.Spec.Source.Pod` or `.Spec.Destination.Pod`
 // should be set.
-func (c *Controller) getTargetCaptureDevice(pc *crdv1alpha1.PacketCapture) string {
-	var pod, ns string
+func (c *Controller) getTargetCaptureDevices(pc *crdv1alpha1.PacketCapture) map[crdv1alpha1.CaptureLocation]string {
+	devices := make(map[crdv1alpha1.CaptureLocation]string)
+
+	// First, determine if the source and/or destination Pods are on the current Node.
+	var sourceDevice, destinationDevice string
 	if pc.Spec.Source.Pod != nil {
-		pod = pc.Spec.Source.Pod.Name
-		ns = pc.Spec.Source.Pod.Namespace
-	} else {
-		pod = pc.Spec.Destination.Pod.Name
-		ns = pc.Spec.Destination.Pod.Namespace
+		sourceDevice = c.getPodDevice(pc.Spec.Source.Pod)
+	}
+	if pc.Spec.Destination.Pod != nil {
+		destinationDevice = c.getPodDevice(pc.Spec.Destination.Pod)
+	}
+
+	// Handle the special case where both Pods are on the same Node.
+	// If so, we perform a single capture.
```

Comment:
this is hard to do the validation in CRD and this 'optimization' seems unnecessary to me. for same node cases it's still 2 devices.